Algorithms/Graphs/MST/prim2.cpp

```diff
@@ -0,0 +1,94 @@
+
+/*  The following is an implementation of Prim's algorithm
+    using adjacency list representation of a graph. The data structure
+    min heap is used for this. There are, again, two implementations
+    in this, the min heap code can be written differently in a class, or
+    the priority_queue present in STL can be used. Here, it's the
+    latter.
+    Now, a priority_queue is essentially a max heap, but can be used
+    as given below to form a min heap. The loop statements execute
+    v+e times, in addition to log(v) time for adding to priority_queue.
+    Overall, O(v+e)*O(log(v)) = O((e+v)*log(v)) = O(e*log(v))
+    Since for a connected graph, v = O(e).
+*/
+
+#include<bits/stdc++.h>
+#define INF INT_MAX
```

Comment:
This macro is unused in the program. Refer line 52. 
Use macros only when extremely essential or solving problems in contests.